Riot/AppDelegate.m

```diff
@@ -1097,6 +1113,55 @@ - (void)application:(UIApplication *)application didRegisterUserNotificationSett
     }
 }
 
+- (void)application:(UIApplication *)application handleActionWithIdentifier:(NSString *)identifier forLocalNotification:(UILocalNotification *)notification withResponseInfo:(NSDictionary *)responseInfo completionHandler:(void (^)())completionHandler
+{
+    if ([identifier isEqualToString: @"inline-reply"])
+    {
+        NSString* roomId = notification.userInfo[@"room_id"];
+        if (roomId.length)
+        {
+            NSArray* mxAccounts = [MXKAccountManager sharedManager].activeAccounts;
+            MXKRoomDataSource* roomDataSource = nil;
+            for (MXKAccount* account in mxAccounts)
+            {
+                MXKRoomDataSourceManager* manager = [MXKRoomDataSourceManager sharedManagerForMatrixSession:account.mxSession];
+                if (manager)
+                {
+                    roomDataSource = [manager roomDataSourceForRoom:roomId create:false];
```

Comment:
We need `create:YES` to make sure we will have an instance of `MXKRoomDataSource`.
But we cannot call it in this loop because we will have instance even if the MXSession does not know the room.

So we need to come back to the previous lookup with `room = [account.mxSession roomWithRoomId:roomId];`. Something like
```
            MXKRoomDataSource* roomDataSource = nil;
            for (MXKAccount* account in mxAccounts)
            {
                MXRoom *room = [account.mxSession roomWithRoomId:roomId];
                if (room)
                {
                    MXKRoomDataSourceManager* manager = [MXKRoomDataSourceManager sharedManagerForMatrixSession:account.mxSession];
                    if (manager)
                    {
                        roomDataSource = [manager roomDataSourceForRoom:roomId create:YES];
                    }
                    break;
                }
            }
```